src/pybind/nnet3/nnet_nnet_pybind_test.py

```diff
@@ -3,38 +3,155 @@
+                self.assertEqual(comp_type, 'BatchNormComponent')
+                # TODO BN get mean and var
+            elif comp_name == 'prefinal-chain.linear':
+                self.assertEqual(comp_type, 'LinearComponent')
+                params = from_dlpack(component.Params().to_dlpack())
+                self.assertEqual(linear_params.shape, (16, 4))
+            elif comp_name == 'prefinal-chain.batchnorm2':
+                self.assertEqual(comp_type, 'BatchNormComponent')
+                # TODO BN get mean and var
+            elif comp_name == 'output.affine':
+                self.assertEqual(comp_type, 'NaturalGradientAffineComponent')
+                linear_params = from_dlpack(
+                    component.LinearParams().to_dlpack())
+                bias_params = from_dlpack(component.BiasParams().to_dlpack())
+                self.assertEqual(linear_params.shape, (3448, 4))
+                self.assertEqual(bias_params.shape, (3448,))
+            elif comp_name == 'prefinal-xent.affine':
+                self.assertEqual(comp_type, 'NaturalGradientAffineComponent')
+                linear_params = from_dlpack(
+                    component.LinearParams().to_dlpack())
+                bias_params = from_dlpack(component.BiasParams().to_dlpack())
+                self.assertEqual(linear_params.shape, (16, 4))
+                self.assertEqual(bias_params.shape, (16,))
+            elif comp_name == 'prefinal-xent.batchnorm1':
+                self.assertEqual(comp_type, 'BatchNormComponent')
+                # TODO BN get mean and var
+            elif comp_name == 'prefinal-xent.linear':
+                self.assertEqual(comp_type, 'LinearComponent')
+                params = from_dlpack(component.Params().to_dlpack())
+                self.assertEqual(linear_params.shape, (16, 4))
+            elif comp_name == 'prefinal-xent.batchnorm2':
+                self.assertEqual(comp_type, 'BatchNormComponent')
+                # TODO BN get mean and var
+            elif comp_name == 'output-xent.affine':
+                self.assertEqual(comp_type, 'NaturalGradientAffineComponent')
+                linear_params = from_dlpack(
+                    component.LinearParams().to_dlpack())
                 bias_params = from_dlpack(component.BiasParams().to_dlpack())
-                print(linear_params.shape)
-            elif "Batch" in comp_type:
-                # stats_sum = from_dlpack(component.StatsSum().to_dlpack())
-                # stats_sumsq = from_dlpack(component.StatsSumsq().to_dlpack())
-                # print(stats_sum.shape)
-                pass
-            elif "LinearComponent" == comp_type:
-                linear_params = from_dlpack(component.LinearParams().to_dlpack())
-                print(linear_params.shape)
+                self.assertEqual(linear_params.shape, (3448, 4))
+                self.assertEqual(bias_params.shape, (3448,))
+            
```

Comment:
there should be an `else` statement at the end.

You can place an `assert` statement or raise an exception there.